earthpy/spatial.py

```diff
@@ -888,3 +924,119 @@ def draw_legend(im_ax, titles=None, cmap=None, classes=None, bbox=(1.05, 1)):
+        will be stacked in the order given in this list.
+    out_path : string
+        A path with a file name for the output stacked raster
+         tif file.
+    arr_out : boolean
+        A boolean argument to designate what is returned in the stacked
+        raster tif output.
+
+    Returns
+    -------
+    If arr_out keyword is True:
+        tuple: The first value representing the result of src.read() of the
+        stacked array and the second value
+        representing the result of src.profile of the stacked array.
+    If arr_out keyword is False:
+        str : A path with a file name for the output stacked raster tif file.
+
+    
+    """
+
+    # Throw warning and exit
+    raise Warning("stack_raster_tifs is deprecated. Use stack(). Exiting...")
+    sys.exit()
+
+    # Set default import to read
+    kwds = {"mode": "r"}
+
+    out_dir = os.path.dirname(out_path)
+    writing_to_cwd = out_dir == ""
+    if not os.path.exists(out_dir) and not writing_to_cwd:
+        raise ValueError(
+            "The output directory path that you provided does not exist"
+        )
+
+    if len(band_paths) < 2:
+        raise ValueError(
+            """The list of file paths is empty. You need at least
+                            2 files to create a stack."""
+        )
+    with contextlib.ExitStack() as context:
+        sources = [
+            context.enter_context(rio.open(path, **kwds))
+            for path in band_paths
+        ]
+
+        # TODO: Check that the CRS and TRANSFORM are the same
+        dest_kwargs = sources[0].meta
+        dest_count = sum(src.count for src in sources)
+        dest_kwargs["count"] = dest_count
+
+        if arr_out:
+            # Write stacked gtif file
+            with rio.open(out_path, "w", **dest_kwargs) as dest:
+                _stack(sources, dest)
+            # Read and return array
+            with rio.open(out_path, "r") as src:
+                return src.read(), src.profile
+        else:
+            # Write stacked gtif file
+            with rio.open(out_path, "w", **dest_kwargs) as dest:
+                return _stack(sources, dest)
+
+
+# Function to be submitted to rasterio
+# TODO: add unit tests - some are here:
+# https://github.com/mapbox/rasterio/blob/master/rasterio/mask.py
+# This function doesn't stand alone because it writes to an open object called
+# in the other function
+# @deprecate
+def _stack(sources, dest):
+    """Stack a set of bands into a single file.
+
+    Parameters
+    ----------
+    sources : list of rasterio dataset objects
+        A list with paths to the bands you wish to stack. Objects
+        will be stacked in the order provided in this list.
+    dest : a rio.open writable object that will store raster data.
+    """
+
+    # Throw warning and exit
+    raise Warning("_stack is deprecated. Exiting...")
+    sys.exit()
+
+    try:
+        for src in sources:
+            src.profile
+
+    except ValueError as ve:
```

Comment:
similarly this code seems extra too given the deprecation

Reply:
@mbjoseph would you agree that we shold remove deprecated code and just have a placeholder for the function so it doesn't impact our testing coverage?

Reply:
removed deprecated code.